tutorials/arvo/ames.md

```diff
@@ -112,40 +115,299 @@ ducts as unique keys.  Stop tracking a peer by sending Ames a
+and encapsulated into
+packets and sent along a flow. The message will be considered successfully
+received once the sender has received the appropriate set of acks in response,
+defined as follows.
+
+There are two types of acks: fragment acks and message acks. Acks are
+not considered messages, and thus are not `%plea`s or `%boon`s, and so are not
+required to appear in a particular order. Given a message split into N
+fragments, the sender of the message will expect N-1 fragment acks followed
+by exactly one message ack. This is because the receiver
+will send a fragment ack for the first N-1 packets it received, and what
+would have been the final fragment ack will instead be a message ack.
+
+Acks are considered to be part of the flow in which that `%plea` or
+`%boon` lives, as the packets containing their fragments and packets acking the
+receipt of those packets are considered to be what makes up a given message.
+Thus a message-level ack must be received before the next message on the flow
+can begin. The full story is more complicated than this, see the section on
+[flows](#flows).
+
+
+#### Nacks
+
+A nack indicates a negative acknowledgement to a `%plea`, meaning that the
+requested action was not performed.
+
+`%boon`s and naxplanations are never nacked.
+Individual packets are never nacked, only complete `%plea` messages. This works
+because only complete packets ever make it to Ames, as malformed
+packets will be discarded by the Ames I/O module in the
+[King](#the-serf-and-the-king). Therefore there is no need for Ames to do
+something like nack a packet whose checksum does not match the `+mug` of the
+body.
+
+A nack will be accompanied by a naxplanation, which is a third type of message
+separate from `%plea`s and `%boon`s. Ames won't give the vane that requested the
+initial `%plea` a nack until it also receives the naxplanation, which it will send to
+the vane as a `%done` gift. Naxplanations may only be acked, never nacked.
+
+#### (N)ack packets
+
+The contents of a given (n)ack packet are deterministic. The `content` (i.e. the
+encrypted portion of the packet) of a fragment ack is obtained by encrypting the `+jam` of the
+following noun:
+
+```hoon
+[our-life her-life bone message-num fragment-num]
+```
+Each datum in this noun is an atom with the aura `@ud` or an aura that nests
+under `@ud`.
+
+Here, `our-life` refers to the [`life`](@/docs/glossary/breach), or revision
+number, of the acking ship's networking keys, and `her-life` is the `life` of
+the ack-receiving ship's networking keys. `bone` is an opaque number identifying the flow, `message-num` denotes the number of the
+message in the flow identified by `bone`, and
+`fragment-num` denotes the number of the fragment of the message identified by
+`message-num` being acked. 
+
+A message (n)ack is an extended fragment ack that is obtained by encrypting the
+`+jam` of the following noun:
+
+```hoon
+[our-life her-life bone message-num fragment-num ok=? lag=@dr]
```

Comment:
There's no `fragment-num` in a message ack.  The ack packet attests to having processed the whole message, so it doesn't need to refer to any particular fragment.